core/action/action.go

```diff
@@ -131,6 +133,10 @@ func (action *Action) Execute(pair *models.RequestResponsePair) error {
 	//adding 200 ms to include some buffer for it to return response
 	time.Sleep(time.Duration(200+action.DelayInMs) * time.Millisecond)
 
+	journalID := <-journalIDChannel
+	close(journalIDChannel)
```

Comment:
it's not recommended by Go to close the channel by the receiver: 
```
// The close built-in function closes a channel, which must be either
// bidirectional or send-only. It should be executed only by the sender,
// never the receiver, and has the effect of shutting down the channel after
// the last sent value is received. 
```
I think that's because only the sender knows when it finishes sending data, and close the channel safely.

Reply:
Yea, makes sense. I thought of doing it at receiver end because we are sending from multiple places(2 places in proxy.go and unit tests) so thought of doing at receiver end.